doc/developer/design/20230223_stabilize_with_mutually_recursive.md

```diff
@@ -0,0 +1,297 @@
+- Feature name: Stabilize `WITH MUTUALLY RECURSIVE`
```

Comment:
@antiguru: I wasn't sure what I should put here. Should this be a kebab-case string that corresponds to the associated feature toggle or a string that gives a name to the feature?

Reply:
I think this is great! Thinking about it, it's probably what would serve as a title for the document.